src/codeu/chat/server/Model.java

```diff
@@ -68,9 +74,95 @@ public int compare(Time a, Time b) {
+    Connection conn = null; // Placeholder for connection to database
+    Statement stmt = null; // Placeholder for sql command
+    try {
+
+      // Loads the JDBC class dynamically to handle the SQL connection
+      // DriverManager cannot create the connection if this class isn't loaded at runtime
+      Class.forName(sqliteClass);
+      
+      // Connect to database. Creates database if it doesn't exist.
+      conn = DriverManager.getConnection("jdbc:sqlite:" + database);
+      DatabaseMetaData dbm = conn.getMetaData();
+      ResultSet rs = dbm.getTables(null, null, userTable, null);
+       
+      if (rs.next()) { // Table exists
+       
+        String sql = "SELECT ID, TIME, NAME FROM " + userTable;
+        stmt = conn.createStatement();
+        ResultSet users = stmt.executeQuery(sql);
+        // loop through the result set
+        while (users.next()) {
+          
+          String username = users.getString("NAME");
+          String uuidString = users.getString("ID");
+          uuidString = uuidString.substring(6, uuidString.length() - 1);
+          
+          try {
+            Uuid useruuid = Uuid.parse(uuidString);
+            User u = new User(useruuid, username, Time.fromMs(users.getLong("TIME"))); 
+            restoreUser(u);
+          } catch (IOException io) {
+              System.out.println(io.getMessage());
+          }
+        }
+
+        stmt.close();
+        conn.close();  
+      } else { // Table does not exist
+          stmt = conn.createStatement();
+          // Command to create USER table
+          String sql = "CREATE TABLE   " + userTable +
+          "(ID TEXT PRIMARY KEY     NOT NULL," +
+          " TIME           BIGINT     NOT NULL, " + 
+          " NAME           TEXT    NOT NULL" + 
+          ")"; 
+          stmt.executeUpdate(sql);
+          stmt.close();
+          conn.close();
+        }
+    } catch ( Exception e ) {
+      System.err.println( e.getClass().getName() + ": " + e.getMessage() );
+      System.exit(0);
+    }
+}
+
+  public void executeSQL(Connection conn, String statement) {
+
+  }
+
   public void add(User user) {
     currentUserGeneration = userGenerations.make();
 
+    Connection conn = null; 
+    Statement stmt = null;
+    try {
+      Class.forName(sqliteClass); 
+      conn = DriverManager.getConnection("jdbc:sqlite:" + database); 
+      stmt = conn.createStatement();
+      String sql = "INSERT INTO " + userTable + " (ID,TIME,NAME) " +  
+                   "VALUES ('"+ user.id.toString() + "', " + user.creation.inMs() + ", '" + user.name + "');"; 
+      stmt.executeUpdate(sql);
+      stmt.close();
+      conn.close();
+    } catch ( Exception e ) {
+        System.err.println( e.getClass().getName() + ": " + e.getMessage() );
+        System.exit(0);
+    }
+    
+    userById.insert(user.id, user);
+    userByTime.insert(user.creation, user);
+    userByText.insert(user.name, user);
+  }
+
+  public void restoreUser(User user) {
```

Comment:
Could you add a javadoc for what this method does and how it should be used? Should this method be public or should it only be called by the Model itself during setup?